### PR TITLE
Add DLR native support part 2

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        ImageClassificationTranslator translator =
+                ImageClassificationTranslator.builder()
+                        .addTransform(new Resize(224, 224))
+                        .addTransform(new ToTensor())
+                        .build();
+        Criteria<Image, Classifications> criteria =
+                Criteria.builder()
+                        .setTypes(Image.class, Classifications.class)
+                        .optTranslator(translator)
+                        .optModelUrls("file:///Users/leecheng/workspace/python/tvm_model_old")
```
Comment (lanking520) [inline dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java:47]: Upload the model to S3

File: dlr/dlr-engine/build.gradle [hunk 2]
```diff
@@ -8,6 +8,12 @@ repositories {
 
 dependencies {
     api project(":api")
+    testImplementation("org.testng:testng:${testng_version}") {
+        exclude group: "junit", module: "junit"
+    }
+    testImplementation "org.slf4j:slf4j-simple:${slf4j_version}"
+    testRuntimeOnly project(":pytorch:pytorch-engine")
```
Comment (lanking520) [inline dlr/dlr-engine/build.gradle:15]: Why import pytorch? Can't DLR just be used alone? If you are looking for adding Hybrid mode, you can always add it back once you finish the Hybrid functionalities

**Review by lanking520 — APPROVED** [review]

**Review by lanking520 — CHANGES_REQUESTED** [review]
need fix

File: dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java (hunk 1 above)
Comment (stu1130) [reply dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java:47]: will address the comment on next PR as TVM model is platform dependent

File: dlr/dlr-engine/build.gradle (hunk 2 above)
Comment (stu1130) [reply dlr/dlr-engine/build.gradle:15]: This is for testing. DLR can't be run alone.

Comment (stu1130) [reply dlr/dlr-engine/build.gradle:15]: In this PR, I added the end2end test so it should come with secondary engine

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+     * @param handle the handle for native DLR model
+     */
+    public DlrSymbolBlock(long handle) {
+        super(handle);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void removeLastBlock() {
+        throw new UnsupportedOperationException("ONNX Runtime not supported");
```
Comment (lanking520) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrSymbolBlock.java:54]: DLR not supported

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/JniUtils.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public static String getDlrVersion() {
+        return DlrLibrary.LIB.getDlrVersion();
+    }
+
+    private static int mapDevice(String deviceType) {
+        if (Device.Type.CPU.equals(deviceType)) {
+            return 1;
+        } else if (Device.Type.GPU.equals(deviceType)) {
```
Comment (lanking520) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/JniUtils.java:79]: Does DLR support multi-GPU?

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/LibUtils.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * </ol>
+ */
+@SuppressWarnings("MissingJavadocMethod")
+public final class LibUtils {
+
+    private LibUtils() {}
+
+    public static void loadLibrary() {
+        // TODO implement
+        System.load("/Users/leecheng/workspace/djl/dlr/dlr-native/libdlr.dylib");
```
Comment (lanking520) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/LibUtils.java:31]: PLease use generic paths.

File: dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java (hunk 1 above)
Comment (lanking520) [reply dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java:48]: The test won'r be valid if we plan to test it on our own environment

Comment (lanking520) [reply dlr/dlr-engine/src/test/java/ai/djl/dlr/integraion/DlrTest.java:48]: Since you mentioned the model is platform dependent. Can we convert the model on the fly? (e.g allow TVM take MXNet model and convert to TVM).

**Review by lanking520 — CHANGES_REQUESTED** [review]

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/JniUtils.java (hunk 4 above)
Comment (stu1130) [reply dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/JniUtils.java:79]: No, you can only create one model on single device

**Review by lanking520 — APPROVED** [review]
